packages/mongo-nav/src/on-element-click-provider/OnElementClickProvider.tsx

```diff
@@ -37,29 +37,12 @@ export default function OnElementClickProvider({
     };
   };
 
-  // const memoizedValue = useMemo(
-  //   (navElement: NavElement, additionalWork?: (e: React.MouseEvent) => void) =>
-  //     wrappedOnElementClick(navElement, additionalWork),
-  //   [onElementClick],
-  // );
-
-  // const wrappedOnElementClick = useMemo(
-  //   (
-  //     navElement: NavElement,
-  //     additionalWork?: (e: React.MouseEvent) => void,
-  //   ) => {
-  //     return (e: React.MouseEvent) => {
-  //       onElementClick(navElement, e);
-  //       if (additionalWork) {
-  //         return additionalWork(e);
-  //       }
-  //     };
-  //   },
-  //   [onElementClick],
-  // );
+  const memoizedWrapper = React.useMemo(() => wrappedOnElementClick, [
```

Comment:
```suggestion
  const memoizedWrapper = React.useMemo(wrappedOnElementClick, [
```

`useMemo` is like `debounce`, it passes on whatever arguments are given to the function returned from memo.

Reply:
👍